apis/core/v1alpha2/core_types.go

```diff
@@ -461,6 +461,7 @@ type DataOutput struct {
 }
 
 // DataInput specifies a data input sink to an object.
+// If input is array, it will be appended to the target field paths.
```

Comment:
what if some scenario need replace the whole array instead of append in the future？

Reply:
In that case we need to specify the patch behavior in data input. But until now I don't any use case for it.